src/components/znkModuleResults/znkModuleResult.service.js

```diff
@@ -0,0 +1,92 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.znkModuleResults').service('ZnkModuleResultsService', [
+        'InfraConfigSrv', '$log', '$q', 'StorageSrv', 'UtilitySrv',
+        function (InfraConfigSrv, $log, $q, StorageSrv, UtilitySrv) {
+
+            var znkModuleResultsService = {};
+            var MODULE_RESULTS_GUIDS_PATH = StorageSrv.variables.appUserSpacePath + '/moduleResults';
+            var MODULE_RESULTS_PATH = 'moduleResults';
+
+            function _isValidNumber(number){
+                if(!angular.isNumber(number) && !angular.isString(number)){
+                    return false;
+                }
+
+                return !isNaN(+number);
+            }
+
+            function _getModuleResultsGuids(){
+                var storage = InfraConfigSrv.getStorageService();
+                return storage.get(MODULE_RESULTS_GUIDS_PATH);
+            }
+
+            function _getInitModuleResult(moduleId, guid){
+                return $q.when( {
+                    "moduleId": moduleId,
+                    "assign": false,
+                    "contentAssign": false,
+                    "date": null,
+                    "tutorId": null,
+                    guid: guid
+                });
+            }
+
+            function _getModuleResultByGuid(guid, moduleId) {
+                var storage = InfraConfigSrv.getStorageService();
+                var path = MODULE_RESULTS_PATH + '/' + guid;
+                return storage.get(path).then(function(moduleResult){
+                    var initResultProm = _getInitModuleResult(moduleId, guid);
+                    return initResultProm.then(function(initResult) {
+                        if(moduleResult.guid !== guid){
```

Comment:
getModuleResultByGuid is called when the user has under 'users/$$uid/moduleResults' a guid for this module id.
the function get the results object, and check if the guid property equals to the guid the user have.
if not, the function override the result object with the init result.
if its equals, the function extend the results Without Override. need to understand from Igor:
1. in what cases the guid property will not be equals to the user-guid.
2. why do we need to extend the results